File: src/hello_world_cpp_pkg/src/simple_pub.cpp
#include "rclcpp/rclcpp.hpp"

#include "std_msgs/msg/string.hpp"

class SimplePublisher : public rclcpp::Node
{
private:
    rclcpp::Publisher<std_msgs::msg::String>::SharedPtr publisher;
    rclcpp::TimerBase::SharedPtr timer;
    int counter = 0;

    void timer_callback()
    {
        auto message = std_msgs::msg::String();
        message.data = "Hello, world! " + std::to_string(counter++);
        this->publisher->publish(message);

        RCLCPP_INFO(this->get_logger(), "Publishing: '%s'", message.data.c_str());
    }


public:
    SimplePublisher() : Node("simple_publisher")
    {
        publisher = this->create_publisher<std_msgs::msg::String>("my_cool_topic", 10);
        timer = this->create_wall_timer(
            std::chrono::seconds(1),
            std::bind(&SimplePublisher::timer_callback, this));
    }
};

int main(int argc, char * argv[])
{
    rclcpp::init(argc, argv);

    auto node = std::make_shared<SimplePublisher>();
    rclcpp::spin(node);

    rclcpp::shutdown();

  return 0;
}

File: src/hello_world_py_pkg/hello_world_py_pkg/simple_pub.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node

from std_msgs.msg import String


class PublisherNode(Node):
    def __init__(self):
        super().__init__('simple_publisher_node')

        self.counter_ = 0

        self.publisher_ = self.create_publisher(String, 'my_cool_topic', 10) # 10 is the queue size
        self.timer_ = self.create_timer(1.0, self.timer_callback)

    def timer_callback(self):
        msg = String()
        msg.data = f'A new message appeared: {self.counter_}'
        self.publisher_.publish(msg)

        self.get_logger().info(f'Publishing: {msg.data}')

        self.counter_ += 1

def main(args=None):
    rclpy.init(args=args)

    node = PublisherNode()
    rclpy.spin(node)

    rclpy.shutdown()

if __name__ == '__main__':
    main()